ci-operator/config/openshift/ceph-csi/openshift-ceph-csi-release-4.7.yaml

```diff
@@ -0,0 +1,19 @@
+build_root:
+  project_image:
+    dockerfile_path: scripts/Dockerfile.test
+resources:
+  '*':
+    limits:
+      memory: 4Gi
+    requests:
+      cpu: 100m
+      memory: 200Mi
+tests:
+- as: static-checks
+  commands: make cephcsi && make test
```

Comment:
these commands do not seem to be running in the image that was build on line 3?

Reply:
What makes you think so? They run in `src` image which is the build root image (=line 3 image) with the source code checked out.

Reply:
My guess is that the check that emits that message is somehow wrong

Reply:
Possibly caused by the fact that `libcephfs-devel` isn't installed?

Reply:
Of course, https://github.com/openshift/ceph-csi/blob/master/scripts/check-env.sh#L44-L46 eats the real error message and emits a generic guess about what's missing.

Reply:
That's correct, we are not installing libcephfs-cevel. I can work on that.
But it also cribs about GO111MODULE and CGO_ENABLED which we are setting [here](https://github.com/ceph/ceph-csi/blob/6c8bc797712537ff1fed4d4948138325465b8b41/scripts/Dockerfile.test#L15)

GO111MODULE should be set to 'on' or 'auto'
failed to initialize build cache at /.cache/go-build: mkdir /.cache: permission denied
CGO_ENABLED should be set to '1'
make: *** [Makefile:130: check-env] Error 2

Reply:
I guess there's probably some other bug somewhere on the path? `src` image is the root image with the working copy of your code checked out, you can inspect both their build logs as well as all pod resources in the collected artifacts.

Reply:
Well, we use the same `Dockerfile.test` in upstream, so the missing `libcephfs-devel` package is fine (good catch though!); it is not required nor used anywhere. Only `librbd-devel` and `librados-devel` are really needed. The error message is incorrect.

The `root` image seems to have set the environment, at least `GO111MODULE`, see `STEP 4` in [root.log](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/15636/rehearse-15636-pull-ci-openshift-ceph-csi-master-static-checks/1369214624959827968/artifacts/build-logs/root.log).

I do not yet understand why/how this environment variable gets removed when adding an additional `src` layer to the image.